core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -842,19 +836,7 @@ public void refresh(String source) {
         verifyNotClosed();
 
         if (canIndex()) {
```

Comment:
empty if clause? also, canIndex always returns true ... so some test should have failed here?

Reply:
oh I am not sure something should have in this case there are tests for the entire thing but they are not very sophisticated. I also think it's very hard to test.